code/model/EditableCustomRule.php

```diff
@@ -147,4 +147,121 @@ public function canUnpublish($member = null)
     {
         return $this->canDelete($member);
     }
+
+	/**
+	 * @param $formFieldWatch
+	 * @param $holderSelector
+	 *
+	 * @return array
+	 */
+	public function BuildExpression($formFieldWatch)
+	{
+		$rule = $this;
+
+		// what action do we need to keep track of. Something nicer here maybe?
+		// @todo encapulsation
+		$action = "change";
+
+		if($formFieldWatch instanceof EditableTextField)
+		{
+			$action = "keyup";
+		}
+
+		// is this field a special option field
+		$checkboxField = false;
+		$radioField = false;
+
+		if(in_array(
+			$formFieldWatch->ClassName, array(
+				'EditableCheckboxGroupField',
+				'EditableCheckbox',
+			)
+		))
+		{
+			$action = "click";
+			$checkboxField = true;
+		}elseif($formFieldWatch->ClassName == "EditableRadioField")
+		{
+			$radioField = true;
+		}
+
+		$target = sprintf('$("%s")',$formFieldWatch->getSelectorFieldOnly());
+
+		// and what should we evaluate
+		switch($rule->ConditionOption)
+		{
+			case 'IsNotBlank':
+				$expression = ($checkboxField || $radioField) ? "{$target}.is(\":checked\")" : "{$target}.val() != ''";
+
+				break;
+			case 'IsBlank':
+				$expression = ($checkboxField || $radioField) ? "!({$target}.is(':checked'))" : "{$target}.val() == ''";
+
+				break;
+			case 'HasValue':
+				if($checkboxField)
+				{
+					$expression = "{$target}.prop('checked')";
+				}elseif($radioField)
+				{
+					// We cannot simply get the value of the radio group, we need to find the checked option first.
+					$expression = sprintf('%s.closest(".field, .control-group").find("input:checked").val()=="%s"',$target,$rule->FieldValue);
+				}else
+				{
+					$expression = sprintf('%s.val() == "%s"',$target,$rule->FieldValue);
+				}
+
+				break;
+			case 'ValueLessThan':
+				$expression = sprintf('%s.val() < parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			case 'ValueLessThanEqual':
+				$expression = sprintf('%s.val() <= parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			case 'ValueGreaterThan':
+				$expression = sprintf('%s.val() > parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			case 'ValueGreaterThanEqual':
+				$expression = sprintf('%s.val() >= parseFloat("%s")',$target,$rule->FieldValue);
+
+				break;
+			default: // ==HasNotValue
+				if($checkboxField)
+				{
+					$expression = sprintf('!%s.prop("checked")',$target);
+				}elseif($radioField)
+				{
+					// We cannot simply get the value of the radio group, we need to find the checked option first.
+					$expression = sprintf('%s.parents(".field, .control-group").find("input:checked").val()!="%s"',$target,$rule->FieldValue );
+				}else
+				{
+					$expression = sprintf('%s.val() != "%s"',$target,$rule->FieldValue);//'$(this).val() != "' . $rule->FieldValue . '"';
+				}
+
+				break;
+		}
+
+		$result = array(
+			'operation'      => $expression,
+			'event'          => $action,
+		);
+		return $result;
+	}
```

Comment:
I feel like this method could be refactored into some smaller, more testable methods